com.ibm.streamsx.topology/opt/python/codegen/py_pyTupleTosplTuple.cgt

```diff
@@ -37,7 +37,19 @@ void MY_OPERATOR::fromPythonToPort<%=$oport->getIndex()%>(PyObject *pyTuple <%=$
     if (<%=$ai%> < frs) {
          // Value from the Python function
          PyObject *pyAttrValue = PyTuple_GET_ITEM(pyTuple, <%=$ai%>);
-         if (!SplpyGeneral::isNone(pyAttrValue)) {
+<%
+    # None in Python tuple is always mapped to default, or matching input.
+    # None in Python dict was remapped temporarily to Python SPL null.
+    if (hasOptionalTypesSupport() && SPL::CodeGen::Type::isOptional($atype)) {
+         # For optional types let dict None map to default value of null.
```

Comment:
I found this code really hard to understand, the comment on 44 doesn't actually tell the full story, in fact it's misleading as there are no dict's here, the None in dict has already been converted to an SPL NULL in a tuple.